core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java

```diff
@@ -1,23 +1,73 @@
 package io.quarkus.deployment.ide;
 
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
 public enum Ide {
 
-    IDEA("idea"),
-    ECLIPSE("eclipse"),
-    VSCODE("code"),
-    NETBEANS("netbeans");
+    IDEA("idea", "--help"),
+    ECLIPSE("eclipse", (String[]) null),
+    VSCODE("code", "--version"),
+    NETBEANS(null);
```

Comment:
Why did you remove netbeans ? Running netbeans will still work if it is in your PATH. (In fact that is how mine works now, as the other way (finding the full path) has broken when I updated to netbeans 12.3, i still need to fix that.

Reply:
So simply running `netbeans ...` will work?
I was under the impression for the previous incarnation of the code that that could never happen for netbeans.

In order for me to fix it, I need to know if `netbeans --version` or `netbeans --help` will work without opening the IDE.
Can you please check and let me know?

Reply:
netbeans --help will give:

```
[pkruger@pkruger ~]$ netbeans --help
Usage: /opt/IDE/netbeans/platform/lib/nbexec {options} arguments

General options:
  --help                show this help 
  --jdkhome <path>      path to Java(TM) 2 SDK, Standard Edition
  -J<jvm_option>        pass <jvm_option> to JVM

  --cp:p <classpath>    prepend <classpath> to classpath
  --cp:a <classpath>    append <classpath> to classpath
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=lcd
Module reload options:
  --reload /path/to/module.jar  install or reinstall a module JAR file

Additional module options:
  --modules                               
  --refresh                               Refresh all catalogs
  --list                                  Prints the list of all modules, their versions and enablement status
  --install <arg1>...<argN>               Installs provided JAR files as modules
  --disable <arg1>...<argN>               Disable modules for specified codebase names
  --enable <arg1>...<argN>                Enable modules for specified codebase names
  --update <arg1>...<argN>                Updates all or specified modules
  --update-all                            Updates all modules
  --extra-uc <arg>                        Add a extra Update Center (URL)
  --start-java-language-server [<arg>]    Starts the Java Language Server
  --start-java-debug-adapter-server <arg> Starts the Java Debug Adapter Server
  --open file1[:line1]...                 open specified file(s), possibly at given location; can also pass project directories
  --open-group NAME                       open a project group by shortened or full name (or unique substring)
  --close-group                           close any open project group
  --list-groups                           list available project groups

Core options:
  --laf <LaF classname> use given LookAndFeel class instead of the default
  --fontsize <size>     set the base font size of the user interface, in points
  --locale <language[:country[:variant]]> use specified locale
  --userdir <path>      use specified directory to store user settings
  --cachedir <path>     use specified directory to store user cache, must be different from userdir
  --nosplash            do not show the splash screen
```

if netbeans is in the PATH

Reply:
the other code can also be a fallback

Reply:
Cool, I'll use that then

Reply:
Fixed